util/gvr.go

```diff
@@ -0,0 +1,73 @@
+package util
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"sigs.k8s.io/yaml"
+)
+
+func readYamlFile(filename string) (*unstructured.Unstructured, error) {
+	yamlFile, err := os.ReadFile(filename)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read yaml file: %w", err)
+	}
+
+	var obj unstructured.Unstructured
+	if err := yaml.Unmarshal(yamlFile, &obj.Object); err != nil {
+		return nil, fmt.Errorf("failed to unmasrhsal into object: %w", err)
+	}
+
+	return &obj, nil
+}
+
+func CreateResource(filename string, dynamicClient *dynamic.DynamicClient, gvr schema.GroupVersionResource, namespace string, log *zap.Logger) error {
+	obj, err := readYamlFile(filename)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve configuration information: %w", err)
+	}
+
+	// Check if resource already exists
+	existing, err := dynamicClient.Resource(gvr).Namespace(namespace).Get(context.TODO(), obj.GetName(), metav1.GetOptions{})
+	if err == nil {
+		log.Info("Resource already exists, skipping creation", zap.String("resource-name", existing.GetName()))
+		return nil
+	} else if !errors.IsNotFound(err) {
+		return fmt.Errorf("failed to check if resource exists: %w", err)
+	}
+
+	result, err := dynamicClient.Resource(gvr).Namespace(namespace).Create(context.TODO(), obj, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to create resource: %w", err)
+	}
+
+	log.Info("Applied resource", zap.String("resource-name", result.GetName()))
+	return nil
+}
+
+func DeleteResourcesWithLabel(dynamicClient *dynamic.DynamicClient, gvr schema.GroupVersionResource, namespace string, labelKey string, labelVal string, log *zap.Logger) error {
```

Comment:
It's almost always good to make a util function as generic as possible so that someone can benefit from it in the future. Can we change `labelKey string, labelVal string` to `labels map[string]string`?

Reply:
i think we already have it quite generic by having labelKey and labelVal rather than just having one variable for label, that way user can define whatever label they want and its corresponding value. The function is meant to delete resources given a specific label, and not multiple labels, so i dont think there is a benefit to having it as a map. WDYT?

Reply:
Yes, what I meant was matching multiple labels. `a=x && b=y`